File: miniapp/src/api/payment.ts
/**
 * 微信支付 API
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/payment'

/** 创建支付请求 */
export interface CreatePaymentReq {
  order_id: string
}

/** 小程序支付参数 */
export interface WeChatPayParams {
  timeStamp: string
  nonceStr: string
  package: string
  signType: string
  paySign: string
  out_trade_no: string
}

/** 创建支付响应 */
export interface CreatePaymentRes {
  success: boolean
  data?: WeChatPayParams
  message: string
}

/** 创建支付 */
export function createPayment(data: CreatePaymentReq) {
  return request<CreatePaymentRes>({
    url: `${PREFIX}/create`,
    method: 'POST',
    data,
  })
}

/** 调起微信支付 */
export function requestWeChatPayment(params: WeChatPayParams): Promise<void> {
  return new Promise((resolve, reject) => {
    uni.requestPayment({
      provider: 'wxpay',
      timeStamp: params.timeStamp,
      nonceStr: params.nonceStr,
      package: params.package,
      signType: params.signType,
      paySign: params.paySign,
      success: () => {
        resolve()
      },
      fail: (err: any) => {
        reject(err)
      },
    })
  })
}

File: backend/app/api/v1/schemas/payment.py
"""
支付相关的 Pydantic schemas
"""
from pydantic import BaseModel


class CreatePaymentRequest(BaseModel):
    """创建支付请求"""

    order_id: str


class PaymentParams(BaseModel):
    """小程序支付参数"""

    timeStamp: str
    nonceStr: str
    package: str
    signType: str
    paySign: str
    out_trade_no: str


class CreatePaymentResponse(BaseModel):
    """创建支付响应"""

    success: bool
    data: PaymentParams | None = None
    message: str


class PaymentErrorResponse(BaseModel):
    """支付错误响应"""

    success: bool
    message: str
